### PR TITLE
Bugfix: Q model and target model used the same weights

### PR DESCRIPTION
**Pull request by mlech26l**
DQN keeps two copies of the weights (a normal Q model for exploration and a target model for computing the target Q values during training).
The code before this pull-request didn't create the two copies **properly**. Instead, the Q model and the target model always shared the same weights.

+ some minor fixes (instantiation of the Huber loss, variable *update_target_network* not used)

### CODE REVIEW

**Comment by fchollet** [comment]
@jacoblchapman please review these changes.

File: examples/rl/deep_q_network_breakout.py [hunk 1]
```diff
@@ -209,13 +214,13 @@
                 q_action = tf.reduce_sum(tf.multiply(q_values, masks), axis=1)
                 # Calculate loss between new Q-value and old Q-value
                 # Clip the deltas using huber loss for stability
-                loss = tf.reduce_mean(keras.losses.huber(updated_q_values, q_action))
+                loss = tf.reduce_mean(keras.losses.Huber()(updated_q_values, q_action))
```
Comment (jacoblchapman) [inline examples/rl/deep_q_network_breakout.py:217]: Maybe I am wrong but I don't think you need to initiate the class?
See here:
https://keras.io/api/losses/regression_losses/#huber-function

Comment (mlech26l) [reply examples/rl/deep_q_network_breakout.py:217]: It should be fine in general, but at least in the release version of TF2.2, the tf.keras.losses.huber function seems to be not exposed [See here](https://github.com/tensorflow/tensorflow/blob/r2.2/tensorflow/python/keras/losses.py).

In the main branch of TF they seem to have added it though [See here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/losses.py#L1421)

Comment (jacoblchapman) [reply examples/rl/deep_q_network_breakout.py:217]: Cool, they will both work :D thanks!

Comment (fchollet) [reply examples/rl/deep_q_network_breakout.py:217]: If you want to use `Huber()`, then:

1. it should be instantiated outside of the loop, and then you should reuse the same instance
2. the `reduce_mean` is no longer necessary
